include/awkward/kernel.h

```diff
@@ -4,11 +4,181 @@
 #define AWKWARD_KERNEL_H_
 
 #include "awkward/common.h"
+#include "awkward/util.h"
+#include "awkward/cpu-kernels/allocators.h"
+#include "awkward/cpu-kernels/getitem.h"
 #include "awkward/cpu-kernels/sorting.h"
 
+#include <sstream>
+
+#ifndef _MSC_VER
+  #include "dlfcn.h"
+#endif
+
+using namespace awkward;
+
 namespace kernel {
+  enum Lib {
+      cpu_kernels,
+      cuda_kernels
+  };
```

Comment:
Enums should usually end with a "num" parameter, like

```c++
enum Lib {
    cpu_kernels,
    cuda_kernels,
    num_libs
};
```

It helps in cases when you might need to write a for loop over all the enums, though I can't think of any reason why one would want to do that with these. It also lets all of the real options end with commas so they're more easily reordered.